test/tests/transact.ts

```diff
@@ -192,55 +137,227 @@ suite('transact', function () {
+            test('triggers', async function () {
+                const {action, session} = await mockData()
+                const result = await session.transact(
+                    {action},
+                    {transactPlugins: [new MockTransactPlugin()]}
+                )
+                assetValidTransactResponse(result)
+            })
+        })
+        suite('transactPluginsOptions', function () {
+            test('transact', async function () {
+                const {action} = await mockData()
+                const session = new Session({
+                    ...mockSessionOptions,
+                    transactPlugins: [new MockTransactResourceProviderPlugin()],
+                })
+                const result = await session.transact(
+                    {action},
+                    {
+                        transactPluginsOptions: {
+                            disableExamplePlugin: true,
+                        },
+                    }
+                )
+                assetValidTransactResponse(result)
+                if (result && result.transaction && result.transaction.actions) {
+                    assert.lengthOf(result.transaction.actions, 1)
+                } else {
+                    assert.fail('Transaction with actions was not returned in result.')
+                }
+            })
+            test('session constructor', async function () {
+                const {action} = await mockData()
+                const session = new Session({
+                    ...mockSessionOptions,
+                    transactPluginsOptions: {
+                        disableExamplePlugin: true,
+                    },
+                    transactPlugins: [new MockTransactResourceProviderPlugin()],
+                })
+                const result = await session.transact({action})
+                assetValidTransactResponse(result)
+                if (result && result.transaction && result.transaction.actions) {
+                    assert.lengthOf(result.transaction.actions, 1)
+                } else {
+                    assert.fail('Transaction with actions was not returned in result.')
+                }
+            })
+            test('kit constructor', async function () {
+                const {action} = await mockData()
+                const sessionKit = new SessionKit({
+                    appName: 'demo.app',
+                    chains: [
+                        {
+                            id: '2a02a0053e5a8cf73a56ba0fda11e4d92e0238a4a2aa74fccf46d5a910746840',
+                            url: 'https://jungle3.greymass.com',
+                        },
+                    ],
+                    fetch,
+                    transactPlugins: [new MockTransactResourceProviderPlugin()],
+                    transactPluginsOptions: {
+                        disableExamplePlugin: true,
+                    },
+                    walletPlugins: [makeWallet()],
+                })
```

Comment:
The `transactPluginOptions` can also be passed into the `SessionKit`, which will set them for all `Session` instances it creates.